File: 0014-Longest-Common-Prefix/solution.c
char* longestCommonPrefix(char** strs, int strsSize) {
    if (strsSize == 0) {
        return ("");
    }
    
    char* first = strs[0];

    for (int i = 0; first[i] != '\0'; i++) {
        for (int j = 1; j < strsSize; j++) {
            if (strs[j][j] == '\0' || strs[j][i] != first[i]) {
                first[i] = '\0';
                return first;
            }
        }
    }
    return first;
}

File: 0014-Longest-Common-Prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        first = strs[0]

        for i in range(len(first)): 
            for j in strs[1:]: 
                if i >= len(j) or j[i] != first[i]:
                    return first[:i]

        return first
